src/main/java/me/zeroeightsix/fiber/tree/ConfigNode.java

```diff
@@ -17,6 +20,44 @@
      */
     String getName();
 
+    /**
+     * Returns the map describing the attributes of this node.
+     *
+     * <p> Attributes store metadata pertaining to the node itself, rather than its value.
+     * The returned map can be mutated by third parties to supplement the default node metadata.
+     * Examples of attributes include translation keys or rendering information.
+     *
+     * <p> As the returned data structure is shared by every attribute source,
+     * attributes should be grouped by namespace.
+     *
+     * @return this node's configurable attributes
+     */
+    Map<Identifier, ConfigAttribute<?>> getAttributes();
```

Comment:
`Identifier`s aren't strings and shouldn't be treated as such! Even if you create two identical `Identifier`s, you can store both in this map and have them point to different things.

This'll produce some unintended side-effects:
```java
attributes.put(new Identifier("fiber", "identifier"), someAttribute);

// Later, we want to get the 'fiber:identifier' attribute...
attributes.get(new Identifier("fiber", "identifier")) == null; // true 
```

Instead, consider
* Having `String` keys instead
* Changing the `getAttributeValue` logic to look for a matching identifier instead, and not the exact same identifier

Reply:
What prevents us from providing `Identifier#equals` ?

Reply:
Absolutely nothing and I'm a dork